_posts/2021-09-29-user-defined-value-types-bug.md

```diff
@@ -0,0 +1,63 @@
+---
+layout: post
+published: true
+title: 'User Defined Value Types Bug'
+date: '2021-09-29'
+author: Solidity Team
+category: Security Alerts
+---
+
+On September 28th, 2021, Harry Altman ([@haltman-at](https://github.com/haltman-at)) of Truffle
+discovered a bug in user defined value types.
+
+The bug has no influence on the correctness of Solidity contracts, but contracts compiled with
+Solidity 0.8.8 that use the new feature are unnecessarily wasteful and might have problems with
+tooling or contract upgrades.
+
+The bug exists only in Solidity 0.8.8 and is fixed in 0.8.9.
+
+We assigned the bug a severity of "very low".
+
+## Storage Layout of User Defined Value Types
+
+The compiler did not correctly compute the storage layout of user defined value types based on types
+that are shorter than 32 bytes. It would always use a full storage slot for these types, even if the
+underlying type is shorter. A user defined value type should behave exactly as its underlying type
+in all aspects apart from additional safety requirements.
```

Comment:
Not sure about the wording for "apart from ...". Looking for suggestions.

Reply:
"A user defined value type in 0.8.8 would behave exactly as its underlying type in all aspects with the exception of storage layout."?

Reply:
'should have the same data representation'?

Reply:
Changed it.